backend/src/main/kotlin/org/genspectrum/dashboardsbackend/model/SubscriptionModel.kt

```diff
@@ -39,11 +44,13 @@ class SubscriptionModel(
             organism = request.organism.name
             active = true
             conditionsMet = false
+            this.userId = userId
         }
         .toSubscription()
 
-    fun deleteSubscription(id: String) {
+    fun deleteSubscription(id: String, userId: String) {
         val subscription = SubscriptionEntity.findById(convertToUuid(id))
+            .takeIf { it?.userId == userId }
```

Comment:
Maybe a method on the entity that bundles `.findById(convertToUuid(id)).takeIf { it?.userId == userId }` makes sense? (Anticipating that the PUT endpoint will also need it)